File: p_practice/Lv2/전화번호목록.js
/**
 * @param {array} phone_book
 */

function solution(phone_book) {
  let answer = true;

  for (let i = 0; i < phone_book.length; i++) {
    phone_book.map(phone => {
      if (phone.startsWith(phone_book[i]) && phone !== phone_book[i]) {
        answer = false;
      }
    });
  }
  console.log(answer);
  return answer;
}

// test
solution(["119", "97674223", "1195524421"]); // return False
solution(["123", "456", "789"]); // return True
solution(["12", "123", "1235", "567", "88"]); // return False

File: programmers/practice/Lv2/전화번호목록.py
# 시간 초과 코드.. 수정 필요
def solution(phone_book):
    answer = True
    for i in range(0,len(phone_book)):
        for phone in phone_book:
            if phone.startswith(phone_book[i]) and phone!= phone_book[i]:
                answer=False;
    print(answer)
    return answer